docs/KFSERVING_DEBUG_GUIDE.md

```diff
@@ -0,0 +1,335 @@
+# Debug KFServing InferenceService Status
+You deployed an InferenceService to KFServing, but it is not in ready state. Go through this step by step guide to understand what failed.
```

Comment:
I've seen occasions where the inferenceservice failed to deploy. e.g. 8443 firewall issues. The above line assumes you've got further than this class of error.  Also maybe we should list a few commands that check whether key controllers and hooks are deployed e.g.  kfserving-controller-manager-0 and the webhook `inferenceservice.serving.kubeflow.org`.  Can we give kubectl commands which print out versions of kubeflow, istio and knative as version compatibility